### PR TITLE
CI: Skip tests with users' action required

### PR DESCRIPTION
**Pull request by thopiekar**
Includes all the unmerged PRs I made before, but I can imagine that this needs to be discussed first.
I know the solution about skipping all win32 tests is not optimal, but a first step towards running CI tests.
Future goal: Fixing all failing tests like #220 and let the test return the errors to the CI. Currently, they seem to be ignored.

### CODE REVIEW

File: pywin32_testall.py [hunk 1]
```diff
@@ -62,11 +62,22 @@ def find_and_run(possible_locations, script, cmdline_rest=""):
...
+    parser = argparse.ArgumentParser(description="A script to trigger tests in all subprojects.")
+    parser.add_argument("-no-user-interaction",
+                        default=False,
+                        action='store_true',
+                        help="Configure the Python environment correctly for pywin32.")
+
+    args = parser.parse_args()
+
+    if not args.no_user_interaction:
+        # win32
```
Comment (mhammond) [inline pywin32_testall.py]: Note that most of the win32 test suite runs without interaction, and testall.py already has a list of tests that require interaction - so you probably want to pass the same cmdline arg to testall.py

Comment (thopiekar) [reply pywin32_testall.py]: I took a look into testall and it seems that it should already skip these tests?
Don't see clearly the point where the switch should be added :confused:

Comment (mhammond) [reply pywin32_testall.py]: > Don't see clearly the point where the switch should be added

Yeah, me either :( Apart from directsound, what issues are you having?

**Comment by thopiekar** [comment]
I believe it works as you expect now :wink:

File: pywin32_testall.py (hunk 1 above)
Comment (thopiekar) [reply pywin32_testall.py]: Well, at the moment all errors are displayed but ignored. I guess the next step will be to make the tests really fail in Appveyor.
So, if you accept, we can run all tests at least, which I think is a great step forward :)

**Comment by mhammond** [comment]
Where do these test results show up? eg, https://ci.appveyor.com/project/mhammond/pywin32/build/job/s8odnwhj3jijed0t/tests says "The build job has not produced any test results."

**Comment by thopiekar** [comment]
As far as I understand from the documentation, you can upload results in XML format. Doesn't seem like you can pipe the output somewhere and let it get parsed by AppVeyor.
I'm not an AppVeyor expert, I can be wrong through.

**Comment by mhammond** [comment]
Sorry, but I'm confused - what exactly is this PR for? It looks like it is to prevent some tests from failing, but I can't see where those failures are currently reported.

**Comment by thopiekar** [comment]
Well, the problem is that the tests are executed, but the return code is not properly passed.
I even added this line to our Appveyor file and the errorcode is still ignored:
https://github.com/thopiekar/pywin32/commit/34622ebff96491774368be4c5f86085a9fc30fe2
You can find the output of the tests in the large output:
https://ci.appveyor.com/project/thopiekar/pywin32/build/job/6beu8nkb25algrth?fullLog=true#L2901

**Comment by thopiekar** [comment]
Found the missing piece which makes the tests fail correctly.
As you can see in the commit message, the return code was never redirected and therefore the Appveyor always received a return code == 0, which is a successful exit of course.

**Comment by thopiekar** [comment]
@mhammond is there anything left that I missed correcting? Any other concerns?

File: pywin32_testall.py [hunk 2]
```diff
@@ -2,54 +2,29 @@
...
-        # some tests prefer to be run from their directory.
-        cmd = [sys.executable, "-u", scriptname] + cmdline_rest.split()
-        print script
-        popen = subprocess.Popen(cmd, shell=True, cwd=dirname,
-                                 stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
-        data = popen.communicate()[0]
-        sys.stdout.buffer.write(data)
-        if popen.returncode:
-            print "****** %s failed: %s" % (script, popen.returncode)
+# a subprocess version - but we prefer the popen one if we can as we can
```
Comment (mhammond) [inline pywin32_testall.py]: This comment is now stale - can you please change it to something like "We prefer popen over subprocess because we can ..."?

Comment (thopiekar) [reply pywin32_testall.py]: Done

**Comment by thopiekar** [comment]
No problem :)

**Pull request merged**